### PR TITLE
chore: update e2e to prove usde changes fixed issue, and add comments to USDE config

### PR DESCRIPTION
**Pull request by juliev0**
### Modifications

Updating the e2e test to prove that [this](https://github.com/numaproj/numaplane/issues/330) issue was fixed.

Also, adding comments to explain the fields in the usde_config.yaml

### CODE REVIEW

File: config/manager/usde-config.yaml [hunk 1]
```diff
@@ -47,11 +52,21 @@ data:
...
       includeSubfields: true
-    - path: spec.sideInputs
+    - path: spec.sideInputs.container.image
+    - path: spec.sideInputs.container.command
+    - path: spec.sideInputs.container.args
+    - path: spec.sideInputs.container.env
       includeSubfields: true
-    - path: spec.templates
+    - path: spec.sideInputs.container.envfrom
+      includeSubfields: true
```
Comment (juliev0) [inline config/manager/usde-config.yaml:61]: broke all of this out into the individual fields

Comment (afugazzotto) [reply config/manager/usde-config.yaml:61]: Great idea!

File: config/manager/usde-config.yaml [hunk 2]
```diff
@@ -47,11 +52,21 @@ data:
       includeSubfields: true
     - path: spec.vertices.udf.builtin
       includeSubfields: true
-    - path: spec.sideInputs
+    - path: spec.sideInputs.container.image
+    - path: spec.sideInputs.container.command
+    - path: spec.sideInputs.container.args
+    - path: spec.sideInputs.container.env
```
Comment (afugazzotto) [inline config/manager/usde-config.yaml:58]: Can you also update the e2e test files related to this:
- https://github.com/numaproj/numaplane/blob/main/tests/manifests/special-cases/ppnd/usde-config.yaml
- https://github.com/numaproj/numaplane/blob/main/tests/manifests/special-cases/progressive/usde-config.yaml

Comment (juliev0) [reply config/manager/usde-config.yaml:58]: thanks for catching! I already forgot about that even though we were just discussing it. :) It sounds like a good motivation for moving the `defaultUpgradeStrategy` out!

Comment (afugazzotto) [reply config/manager/usde-config.yaml:58]: Working on that now, should be ready soon

**Pull request merged**